lib/rust/ensogl/core/src/debug/monitor.rs

```diff
@@ -896,7 +848,8 @@ macro_rules! stats_sampler {
                 $label
             }
             fn value(&self) -> f64 {
-                self.stats.$stats_method() as f64 / $value_divisor
+                let raw_value: f64 = self.stats.$stats_method().as_();
+                raw_value / $value_divisor
```

Comment:
why this change?

Reply:
We're introducing `f64` stats in this PR (`fps`, `frame_time`) - in those cases, `$stats_method()` evaluates to `f64`, and `f64 as f64` results in a warning about trivial numeric cast. We have the following options to work around this warning:

 - a) Keep the code as it was, but add `#![allow(trivial_numeric_casts)]` above, covering the whole expression, and explain in a comment why we use it;
 - b) Split out just the `self.stats.$stats_method() as f64` to a separate helper variable, and add `#![allow(trivial_numeric_casts)]` above it and add a comment with explanation;
 - c) Use `.as_()` provided by `num_traits::cast::AsPrimitive`, which works correctly (identity transformation) for `f64` without emitting a warning, which can be seen as an explicit statement of intent. Though, we could also add a comment explaining the reason why we're using it above, if you believe that's preferable.

Notably, we can't use `.into()` or `.try_into()` here, as they have different semantics for conversion of integer types (esp. `u64` and `usize`) into f64 vs. `as f64`. Specifically, `as f64` (and `.as_()`) perform an approximation if the converted type cannot fit into `f64` without loss of precision - this behavior sounds reasonable for use in stats - whereas `.try_into()` returns an error, forcing us to do approximation (or any other handling) by some other means.

Reply:
Perfect! I'm feeling right now bad that you spent so much time on explaining this to me. Thank you for doing that, the explanation is way more amazing that I was expecting it to be! The solution makes perfect sense and I also think it's the best out of the options!